transformations/auth.py

```diff
@@ -14,42 +19,69 @@ def login():
-        try:
-            ldap_client.protocol_version = ldap.VERSION3
-            ldap_client.simple_bind_s(user_dn, password)
-        except Exception as ex:
-            error = ex
+        user = None
+        if re.search(r'LDAP', current_app.config["AUTH_TYPE"], flags=re.IGNORECASE) or "LDAP_HOSTNAME" \
+                                                                            in current_app.config.keys():
+            if "AUTH_HOSTNAME" in current_app.config.keys():
+                ldap_hostname = current_app.config['AUTH_HOSTNAME']
+            else:
+                ldap_hostname = current_app.config['LDAP_HOSTNAME']
+            ldap_client = ldap.initialize(ldap_hostname)
+            ldap_client.set_option(ldap.OPT_REFERRALS, 0)
+            user_dn = "uid=%s,%s,%s" % (username, current_app.config['LDAP_USER_DN'], current_app.config['LDAP_BASE_DN'])
+            group_dn = "%s,%s" % (current_app.config['LDAP_GROUP_DN'], current_app.config['LDAP_BASE_DN'])
+            try:
+                ldap_client.protocol_version = ldap.VERSION3
+                ldap_client.simple_bind_s(user_dn, password)
+            except Exception as ex:
+                error = ex
 
-        try:
+            try:
 
-            search_scope = ldap.SCOPE_SUBTREE
-            search_filter = "(&(objectClass=%s)(memberOf=cn=%s,%s)(uid=%s))" % (current_app.config['LDAP_OBJECTCLASS'],
-                                                                                current_app.config['LDAP_GROUP'],
-                                                                                group_dn, username)
-            ldap_result = ldap_client.search_s(current_app.config['LDAP_BASE_DN'], search_scope, search_filter)
-            if 0 == len(ldap_result):
-                error = "cannot find in this group"
+                search_scope = ldap.SCOPE_SUBTREE
+                search_filter = "(&(objectClass=%s)(memberOf=cn=%s,%s)(uid=%s))" % (current_app.config['LDAP_OBJECTCLASS'],
+                                                                                    current_app.config['LDAP_GROUP'],
+                                                                                    group_dn, username)
+                ldap_result = ldap_client.search_s(current_app.config['LDAP_BASE_DN'], search_scope, search_filter)
+                if 0 == len(ldap_result):
+                    error = "cannot find in this group"
 
-        except Exception as ex:
-            error = ex
+            except Exception as ex:
+                error = ex
 
-        ldap_client.unbind_s()
-        user = dict()
+            ldap_client.unbind_s()
+            user = dict()
+        elif re.search(r'Clowder', current_app.config["AUTH_TYPE"], flags=re.IGNORECASE):
+            auth_hostname = current_app.config["AUTH_HOSTNAME"]
+            if "SSL_CERT" not in current_app.config.keys() or current_app.config["SSL_CERT"] is None \
```

Comment:
why not use if url.startswith("https://") use HTTPSConnection and then if exist SSL_CERT use that.

Reply:
main problem with the shorter and shorter time of certs this will result in a lot of updates.

Reply:
Made this change. Made perfect sense to not require SSL_CERT, and use the system version unless specified.